draco/analysis/flagging.py

```diff
@@ -1687,3 +1687,66 @@ def process(self, data):
             dset *= tools.invert_no_zero(weight)
 
         return data
+
+
+class ThresholdVisWeightBaseline(task.SingleTask):
+    """Set any weight less than the user specified threshold equal to zero.
+
+    Threshold is determined as `maximum(absolute_threshold,
+    relative_threshold * mean(weight))` and is evaluated per product/stack
+    entry. This cannot be represented as an RFI mask and so the results are
+    applied in place.
+
+    This is the same as the previous version as `ThresholdVisWeight`.
+
+    Parameters
+    ----------
+    absolute_threshold : float
+        Any weights with values less than this number will be set to zero.
+    relative_threshold : float
+        Any weights with values less than this number times the average weight
+        will be set to zero.
+    """
+
+    absolute_threshold = config.Property(proptype=float, default=1e-7)
+    relative_threshold = config.Property(proptype=float, default=0.0)
+
+    def process(self, timestream):
+        """Apply threshold to `weight` dataset.
+        Parameters
+        ----------
+        timestream : `.core.container` with `weight` attribute
+        Returns
+        -------
+        timestream : same as input timestream
+            The input container with modified weights.
+        """
+        from mpi4py import MPI
+
+        timestream.redistribute(["prod", "stack"])
+
+        weight = timestream.weight[:]
+
+        # Average over the frequency and time axes to get a per baseline
+        # average
+        mean_weight = weight.mean(axis=2).mean(axis=0)
```

Comment:
If some weights are already zero, should we skip over these when taking the means? Something like:
```
# Average over time axis, excluding weight-zero samples from the average
mean_weight = np.nan_to_num(np.apply_along_axis(lambda x: np.mean(x[x != 0]), 2, weight))
mean_weight = np.nan_to_num(np.apply_along_axis(lambda x: np.mean(x[x != 0]), 0, mean_weight))
```

Reply:
I think modifying this isn't something to consider at this time. I'm just trying to reintroduce the old thresholding mechanism here which we know works. In the future we should definitely look into a scheme to replace this task.

Reply:
That is another known issue with this task though, and it is already addressed in the new `ThresholdVisWeight`.